ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/TagForm/TagFormChanges/TagFormChanges.tsx

```diff
@@ -164,32 +187,55 @@ export const TagFormChanges = ({ machines }: Props): JSX.Element | null => {
             "positive",
             (tag) => {
               setFieldValue(
-                "tags",
-                values.tags.filter((id) => tag.id !== toFormikNumber(id))
+                "added",
+                values.added.filter((id) => tag.id !== toFormikNumber(id))
               );
             },
             <>
-              <span className="u-nudge-left--small">Discard</span>
+              <span className="u-nudge-left--small">{Label.Discard}</span>
               <Icon name="close" />
             </>
           ),
           ...generateRows(
+            TestId.Removed,
+            removedTags,
+            machineCount,
+            tagIdsAndCounts,
+            Label.Removed,
+            toggleTagDetails,
+            "negative",
+            (tag) => {
+              setFieldValue(
+                "removed",
+                values.removed.filter((id) => tag.id !== toFormikNumber(id))
+              );
+            },
+            <>
+              <span className="u-nudge-left--small">{Label.Discard}</span>
+              <Icon name="close" />
```

Comment:
Could you add `aria-hidden=true` for this icon, this is redundant as we have an appropriate label for this action. 

You can see how this adds noise on the screenshot below (VoiceOver announces this button as a group because of the icon).
![image](https://user-images.githubusercontent.com/7452681/161503949-e8ab3eaf-747c-4a30-88ff-4387d24793e1.png)